src/legacy/core_plugins/console/public/np_ready/application/containers/editor/legacy/console_editor/editor.tsx

```diff
@@ -81,10 +86,54 @@ function EditorUI() {
   useEffect(() => {
     editorInstanceRef.current = senseEditor.create(editorRef.current!);
 
-    const { content: text } = history.getSavedEditorState() || {
-      content: DEFAULT_INPUT_VALUE,
+    const getQueryParams = () => {
+      const [, queryString] = window.location.hash.split('?');
+      return qs.parse(queryString || '');
     };
-    editorInstanceRef.current.update(text);
+
+    const onHashChange = debounce(() => {
+      const qParams = getQueryParams();
+      if (!qParams || !qParams.load_from) {
+        return;
+      }
+      const sourceLocation: string = qParams.load_from;
+      if (/^https?:\/\//.test(sourceLocation)) {
+        const loadFrom: Record<string, any> = {
+          url: sourceLocation,
+          // Having dataType here is required as it doesn't allow jQuery to `eval` content
+          // coming from the external source thereby preventing XSS attack.
+          dataType: 'text',
+          kbnXsrfToken: false,
+        };
+
+        if (/https?:\/\/api\.github\.com/.test(sourceLocation)) {
+          loadFrom.headers = { Accept: 'application/vnd.github.v3.raw' };
+        }
+
+        $.ajax(loadFrom).done(async data => {
+          await editorInstanceRef.current!.update(data);
```

Comment:
The behavior I'm seeing is that the loaded data overwrites everything in the request panel. Is this intended? Perhaps we should prepend it to the saved requests instead?

Reply:
It looks like this is consistent with legacy behaviour:

Starting here: https://github.com/elastic/kibana/blob/44a4cee0da4def92176a9b5a31ec95a82da28644/src/legacy/core_plugins/console/public/src/app.js#L82

Which would ultimately call:
https://github.com/elastic/kibana/blob/44a4cee0da4def92176a9b5a31ec95a82da28644/src/legacy/core_plugins/console/public/src/sense_editor/editor.js#L170

Which sets the value for the entire current buffer.

I agree that changing the current behaviour to rather prepend would be better, but I think once we have multiple buffers we can deal with this in a more elegant manner, perhaps with a dedicated "view in console" buffer - I wouldn't want to change the legacy behaviour in this fix. Wdyt?

Reply:
> I wouldn't want to change the legacy behaviour in this fix.

I agree!